micro_sam/automatic_segmentation.py

```diff
@@ -478,37 +478,36 @@ def _convert_argval(value):
     )
 
     # Run automatic segmentation per image.
-    for input_path in tqdm(input_paths, desc="Run automatic segmentation"):
+    for i, input_path in tqdm(enumerate(input_paths), total=len(input_paths), desc="Run automatic segmentation"):
         if has_one_input:  # if we have only one image / volume.
             _embedding_fpath = embedding_path
-            if args.tracking:
-                _output_fpath = output_path
-            else:
-                _output_fpath = str(Path(output_path).with_suffix(".tif"))
+            _output_fpath = output_path if args.tracking else str(Path(output_path).with_suffix(".tif"))
 
         else:  # if we have multiple image, we need to make the other target filepaths compatible.
+
+            # Decide on the input filename.
+            input_name = f"embeddings_{i:05d}" if args.tracking else Path(input_path).stem
+
             # Let's check for 'embedding_path'.
             _embedding_fpath = embedding_path
             if embedding_path:
-                if _has_extension(embedding_path):  # in this case, use filename as addl. suffix to provided path.
-                    _embedding_fpath = str(Path(embedding_path).with_suffix(".zarr"))
-                    _embedding_fpath = _embedding_fpath.replace(".zarr", f"_{Path(input_path).stem}.zarr")
-                else:   # otherwise, for directory, use image filename for multiple images.
+                if _has_extension(embedding_path):  # Use input filename as addl. suffix to embedding path.
```

Comment:
I think this is really complicated. Let's just remove an extension if there is one and treat the embedding path as the root level folder for saving embeddings, as in the `else` branch.